### ISSUE TITLE
`npm prune --no-shrinkwrap` is acting unexpectedly

### ISSUE THREAD

**Issue by xcambar** [npm/npm#19522]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

`npm prune --no-shrinkwrap` (which option is not documented) produces builds in which *some* packages declared in `dependencies` with the syntax `github:org/repo#commitish` are deleted from `node_modules`.

#### How can the CLI team reproduce the problem?

The following repo will help reproduce the issue: https://github.com/xcambar/npm-prune
The CI https://circleci.com/gh/xcambar/npm-prune/tree/master shows the deletion of the (declared) `lodash` package.

### supporting information:

 - `npm -v` prints:  any npm version > 5.3.0
 - `node -v` prints: any node version > 8
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: MacOS and Ubuntu

**Comment by xcambar** [comment]
tl;dr; `npm prune is defined with: 

> This command removes "extraneous" packages. [...] Extraneous packages are packages that are not listed on the parent package's dependencies list.

This repo (https://github.com/xcambar/npm-prune) has 2 dependencies declared with the `github:org/repo#commitish` syntax. `npm prune --no-shrinkwrap` systematically removes one of them. Which one is removed is environment-dependent.
